addons/pos_self_order/controllers/mobile_entry.py

```diff
@@ -55,12 +50,15 @@ def pos_self_order_start(self, config_id=None, access_token=None, table_identifi
             pos_config_sudo = request.env["pos.config"].sudo().search([
                 ("id", "=", config_id), ("self_order_view_mode", "=", True)], limit=1)
         else:
-            pos_config_sudo = get_any_pos_config_sudo()
+            pos_config_sudo = request.env["pos.config"].sudo().search([
+                ('|'),
+                ("self_order_view_mode", "=", True),
+                ("self_order_kiosk", "=", True)], limit=1)
```

Comment:
restore the `get_any_pos_config_sudo` function. The logic is used in multiple places. To avoid new file, just declare the function in this file.

Also, make sure it's only used in this file.